scipy/sparse/extract.py

```diff
@@ -35,11 +35,11 @@ def find(A):
 
     """
 
-    A = coo_matrix(A).tocsr()  # sums duplicates
-    A.eliminate_zeros()        # removes explicit zeros
-    A = A.tocoo(copy=False)    # (cheaply) convert to COO
-
-    return A.row,A.col,A.data
+    A = coo_matrix(A, copy=False)
```

Comment:
`copy=False` changes the behavior: if `A` was already COO, it gets canonicalized in-place.